app/uk/gov/hmrc/plasticpackagingtax/registration/models/genericregistration/Partner.scala

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright 2022 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.plasticpackagingtax.registration.models.genericregistration
+
+import play.api.libs.json.{Json, OFormat}
+import uk.gov.hmrc.plasticpackagingtax.registration.forms.organisation.PartnerTypeEnum.PartnerTypeEnum
+
+import java.util.UUID
+
+case class Partner(
+  id: String = UUID.randomUUID().toString,
+  partnerType: PartnerTypeEnum,
```

Comment:
Should this have been optional?

Reply:
All partners will have a type. That said more flexible if it is optional but could work if the Partner object is created after selection of the type. On balance, optional is better.

Reply:
Will make optional.